frontend/src/components/BCGrid/BCGridServer.jsx

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+// ag-grid components
+import { ClientSideRowModelModule } from '@ag-grid-community/client-side-row-model'
+import { ModuleRegistry } from '@ag-grid-community/core'
+import { AgGridReact } from '@ag-grid-community/react'
+import '@ag-grid-community/styles/ag-grid.css'
+import '@ag-grid-community/styles/ag-theme-alpine.css'
+// react components
+import { PropTypes } from 'prop-types'
+import { useState, useEffect, useCallback, useMemo } from 'react'
+// api service
+import { useApiService } from '@/services/useApiService'
+// Internal Components
+import Loading from '@/components/Loading'
+// @mui components
+import { TablePagination } from '@mui/material'
+import BCAlert from '@/components/BCAlert'
+import BCBox from '@/components/BCBox'
+// Register the required feature modules with the Grid
+ModuleRegistry.registerModules([ClientSideRowModelModule])
+
+const BCGridServer = (props) => {
+  const defaultGridOptions = {
+    overlayNoRowsTemplate: 'No rows found',
+    autoSizeStrategy: { type: 'fitCellContents' },
+    suppressDragLeaveHidesColumns: true,
+    suppressMovableColumns: false,
+    suppressColumnMoveAnimation: false,
+    rowSelection: 'multiple',
+    animateRows: true,
+    suppressPaginationPanel: true,
+    suppressScrollOnNewData: true,
+    rowHeight: 50
+  }
+  const gridOptions = useMemo(() => ({
+    ...defaultGridOptions,
+    ...props.gridOptions
+  }))
+
+  const [page, setPage] = useState(1)
+  const [size, setSize] = useState(10)
+  const [sortModel, setSortModel] = useState([
+    {
+      field: 'display_name',
+      direction: 'asc'
+    }
+  ])
+  const [filterModel, setFilterModel] = useState([])
+  const [total, setTotal] = useState(0)
+  const [rowData, setRowData] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [isError, setIsError] = useState(false)
+  const [error, setError] = useState('')
+
+  const apiService = useApiService()
+  const fetData = useCallback(
+    () =>
+      apiService({
+        method: 'post',
```

Comment:
I think this will have to be an option. Currently the view is trying to post to the organizations/list get method.  Are you thinking this will always be post? The organization list endpoint will have to be altered.

Reply:
yes, since we need to pass the filter and sort models this needs to be changed to post.